### ISSUE TITLE
1.0.0-beta2 incompatible with react-hot-loader

### ISSUE THREAD

**Issue by globexdesigns** [rackt/react-router#1418]
I use react-hot-loader for development to quickly reload components when they update. The 1.0.0-beta2 build has broken this compatibility. Anytime a module is hot loaded, this message is printed to the console:

```Uncaught Error: Invariant Violation: <Router history> may not be changed```

Any thoughts about how to get around this limitation?

The code that defines the history is pretty standard:

```
export default class App {
	render () {
		return (
			<Router history={new BrowserHistory()}>
				<Route component={AppView} path="/">
					...
				</Route>
			</Router>
		);
	}
}
```

**Comment by Schniz** [comment]
as I stated [here](https://github.com/rackt/react-router/pull/1339#issuecomment-114649224), you should not create a new instance in the props. I haven't used react-hot-reloader yet, but I'm sure there is a place you can define some things that never change. and that's the place you should stick the `new BrowserHistory()`

**Comment by Schniz** [comment]
If I were you, I'd just try to add
```js
const history = new BrowserHistory();
export default class App {
    render () {
        return (
            <Router history={ history }>
                <Route component={AppView} path="/">
                    ...
                </Route>
            </Router>
        );
    }
}
```

**Comment by globexdesigns** [comment]
That did the trick. We should update docs to reflect this.

**Comment by mjackson** [comment]
We merged #1429 which addresses this. Closing.
